src/main/java/com/st/lms/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,73 @@
+package com.st.lms.exception;
+
+import java.sql.SQLException;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.HttpRequestMethodNotSupportedException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+import org.springframework.web.servlet.NoHandlerFoundException;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+	
+	//the custom exceptions
+	@ExceptionHandler({NotFoundException.class, BadRequestException.class, AlreadyExistsException.class})
+	public ResponseEntity<ApiError> handleRegularExceptions(Exception ex, WebRequest request) {
+		
+		if(ex instanceof NotFoundException) {
+			ApiError apiE = new ApiError(404, "NOT FOUND", ex.getMessage());

```

Comment:
It's pointless to define a variable that is then immediately referenced and never referenced again. Inline each of these `apiE` variables into its sole use.